packages/runtime/src/templates/getHandler.ts

```diff
@@ -130,6 +130,14 @@ const makeHandler = ({ conf, app, pageRoot, NextServer, staticManifest = [], mod
     const query = new URLSearchParams(event.queryStringParameters).toString()
     event.path = query ? `${event.path}?${query}` : event.path
 
+    if (event.headers['accept-language']) {
+      // keep just first language to match Netlify redirect limitation:
+      // https://docs.netlify.com/routing/redirects/redirect-options/#redirect-by-country-or-language
+      // > Language-based redirects always match against the first language reported by the browser in the Accept-Language header regardless of quality value weighting.
+      // If we wouldn't keep just first language, it's possible for `next-server` to generate locale redirect that could be cached by ODB
+      event.headers['accept-language'] = event.headers['accept-language'].replace(/\s*,.*$/, '')
```

Comment:
maybe this should be done just for `odb` mode, and let `ssr` keep all languages, but it feels weird to change behavior between 2 of them?

Reply:
Having different behaviour between SSR/ODB feels wrong, and I guess it's because what we're doing is actually trying to workaround our lack of support by imposing the same limitation on Next, which would be a regression.

Reply:
Yeah, to me it's picking one of 2 bad options and figuring it which one is worse/better. I do agree it's probably better to keep SSR pages behaving as they are (so supporting matching on all languages on the header, even if there are multiple source that can produce redirect) and limit the workaround to just ODB (it does technically make things worse to match on lowest common denominator to avoid caching wrong things).

In any case https://github.com/netlify/next-runtime/pull/2138/commits/4c85c21dc8198335612e24cb1556a0e81e18f5c3 limits the workaround just to ODB